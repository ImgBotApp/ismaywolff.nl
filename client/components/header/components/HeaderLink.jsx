import styled from 'styled-components'
import styles from '../../../styles'

const HeaderLink = styled.span`
  & > a {
    text-decoration: none;
    font-family: 'Bitter', serif;
    color: black;
    transition: border-bottom 200ms;
    padding-bottom: 0.15rem;
    font-size: ${props => props.size};
    border-bottom: ${styles.link.inactive};
  }

  & > a.active,
  & > a:hover {
    border-bottom: ${styles.link.active};
    padding-bottom: 0;
  }
`

HeaderLink.defaultProps = {
  size: '1rem'
}

export default HeaderLink
