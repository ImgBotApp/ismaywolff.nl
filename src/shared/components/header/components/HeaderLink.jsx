import styled from 'styled-components'
import { colors } from '../../../styles'

const HeaderLink = styled.span`
  margin-left: 0.5rem;
  margin-right: 0.5rem;
  order: 2;

  & > a {
    color: ${colors.black};
    font-family: Bitter, serif;
    font-size: ${props => props.size};
    text-decoration: none;
  }

  & > a.active,
  & > a:hover {
    text-decoration: underline;
  }

  @media (min-width: 20em) {
    margin-left: 0;
    margin-right: 0;
    order: unset;
  }
`

HeaderLink.defaultProps = {
  size: '1rem'
}

export default HeaderLink
