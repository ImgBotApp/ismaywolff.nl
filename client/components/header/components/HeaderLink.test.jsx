import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { matcher, serializer } from 'jest-glamor-react'
import HeaderLink from './HeaderLink'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<HeaderLink />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HeaderLink />)
    expect(shallowToJson(wrapper)).toMatchSnapshotWithGlamor()
  })

  it('accepts size parameter', () => {
    const wrapper = shallow(<HeaderLink size={'100%'} />)
    expect(shallowToJson(wrapper)).toMatchSnapshotWithGlamor()
  })
})
