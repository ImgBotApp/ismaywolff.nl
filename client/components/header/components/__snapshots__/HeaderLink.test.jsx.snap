exports[`<HeaderLink /> accepts size parameter 1`] = `
<span
  className="fTjLpG"
  size="100%" />
`;

exports[`<HeaderLink /> renders correctly 1`] = `
<span
  className="dOskKx"
  size="1rem" />
`;
